src/ngraph/runtime/gpu/cublas_emitter.cpp

```diff
@@ -0,0 +1,237 @@
+    }
+
+    if (shape_size(arg0_shape) == 0 || shape_size(arg1_shape) == 0)
+    {
+        size_t elemSize = shape_size(out_shape) * dtype.size();
+        dot.reset(new gpu::primitive{[=](void** inputs, void** outputs) {
+            runtime::gpu::cuda_memset(outputs[0], 0, elemSize);
+            debug_sync();
+        }});
+
+        return this->m_primitive_emitter->get_primitive_index(dot, hash);
+    }
+    // case that can be treat as dot1d
+    if ((arg0_shape.size() == arg1_shape.size()) && (arg0_shape.size() == reduction_axes))
+    {
+        for (int i = 0; i < arg0_shape.size(); i++)
+        {
+            if (arg0_shape[i] != arg1_shape[i])
+            {
+                throw std::invalid_argument("arg0 and arg1 shape does not match for dot.");
+            }
+        }
+
+        size_t count = shape_size(arg0_shape);
+        dot.reset(new gpu::primitive{[=](void** inputs, void** outputs) {
+            CUBLAS_SAFE_CALL(cublasSdot(*m_ctx->cublas_handle,
+                                        count,
+                                        static_cast<const float*>(inputs[0]),
+                                        1,
+                                        static_cast<const float*>(inputs[1]),
+                                        1,
+                                        static_cast<float*>(outputs[0])));
+
+            debug_sync();
+        }});
+
+        primitive_index = this->m_primitive_emitter->get_primitive_index(dot, hash);
+    }
+
+    // matrix vector
+    else if ((arg0_shape.size() == 2) && (arg1_shape.size() == 1) && (reduction_axes == 1))
+    {
+        dot.reset(new gpu::primitive{[=](void** inputs, void** outputs) {
+            const float alpha = 1.0;
+            const float beta = 0;
+
+            CUBLAS_SAFE_CALL(cublasSetPointerMode(*m_ctx->cublas_handle, CUBLAS_POINTER_MODE_HOST));
+            CUBLAS_SAFE_CALL(cublasSgemv(*m_ctx->cublas_handle,
+                                         CUBLAS_OP_T,
+                                         arg0_shape[1],
+                                         arg0_shape[0],
+                                         &alpha,
+                                         static_cast<const float*>(inputs[0]),
+                                         arg0_shape[1],
+                                         static_cast<const float*>(inputs[1]),
+                                         1,
+                                         &beta,
+                                         static_cast<float*>(outputs[0]),
+                                         1));
+            CUBLAS_SAFE_CALL(
+                cublasSetPointerMode(*m_ctx->cublas_handle, CUBLAS_POINTER_MODE_DEVICE));
+
+            debug_sync();
+        }});
+
+        primitive_index = this->m_primitive_emitter->get_primitive_index(dot, hash);
+    }
+
+    else
+    {
+        size_t num_of_axes_for_m = arg0_shape.size() - reduction_axes;
```

Comment:
You use count earlier and num_of here. This could be axes_for_m_count or not, you decide.